client/tailscale/localclient.go

```diff
@@ -81,6 +89,31 @@ func (lc *LocalClient) socket() string {
 	return paths.DefaultTailscaledSocket()
 }
 
+// WithRequestHeaders returns a copy of the *LocalClient that will send
+// the given http headers on every request.
+func (lc *LocalClient) WithRequestCallback(f func(*http.Request) error) *LocalClient {
+	return &LocalClient{
+		Dial:             lc.Dial,
+		Socket:           lc.Socket,
+		UseSocketOnly:    lc.UseSocketOnly,
+		tsClient:         lc.tsClient,
+		responseCallback: lc.responseCallback,
+		requestCallback:  f,
+	}
+}
+
+// WithResponseCallback calls f on every incoming response.
+func (lc *LocalClient) WithResponseCallback(f func(*http.Response) error) *LocalClient {
+	return &LocalClient{
+		Dial:             lc.Dial,
+		Socket:           lc.Socket,
+		UseSocketOnly:    lc.UseSocketOnly,
+		tsClient:         lc.tsClient,
+		requestCallback:  lc.requestCallback,
+		responseCallback: f,
+	}
+}
```

Comment:
this seems like a lot of mechanism and API to expose in the package to users. I'd rather not show all this.